Source/Lib/Codec/EbCodingLoop.c

```diff
@@ -3145,11 +3159,11 @@ EB_EXTERN void EncodePass(
     }
 
 
-    EB_BOOL useDeltaQp = (EB_BOOL)(sequenceControlSetPtr->staticConfig.improveSharpness || sequenceControlSetPtr->staticConfig.bitRateReduction);
+    EB_BOOL useDeltaQp = (EB_BOOL)(sequenceControlSetPtr->staticConfig.improveSharpness || sequenceControlSetPtr->staticConfig.bitRateReduction || sequenceControlSetPtr->staticConfig.segmentOvEnabled);
```

Comment:
Is it better to check (pictureControlSetPtr->ParentPcsPtr->segmentOvArray[lcuPtr->index].ovFlags & EB_QP_OV_DELTA)?

Reply:
pictureControlSetPtr->ParentPcsPtr->segmentOvArray could be NULL (memory is allocated only if sequenceControlSetPtr->staticConfig.segmentOvEnabled is True ).  To make code as simple as possible I would leave this kind of change as is

Reply:
So how about
```
EB_BOOL useDeltaQp = (EB_BOOL)(sequenceControlSetPtr->staticConfig.improveSharpness || sequenceControlSetPtr->staticConfig.bitRateReduction ||
    (sequenceControlSetPtr->staticConfig.segmentOvEnabled && (pictureControlSetPtr->ParentPcsPtr->segmentOvArray[lcuPtr->index].ovFlags & EB_QP_OV_DELTA)));
```
?

Because delta QP is subset of your seg OV feature, and this checking would make it more accurate.

Reply:
"EB_BOOL useDeltaQp"  parameter is set when separate blocks("LCU") have different value than frame QP. Those changes works similarly to improveSharpness/bitRateReduction. Even if this is just part of seg OV feature, checking segmentOvEnabled is sufficient and more save.